tools/bin/build_image.sh

```diff
@@ -14,4 +14,21 @@ assert_root
 
 cd "$PROJECT_DIR"
 
-DOCKER_BUILDKIT=1 docker build -f "$DOCKERFILE" . -t "$TAG" --iidfile "$ID_FILE"
+if [[ -z "$CI" ]]; then
+  # run standard build locally (not on CI)
+  DOCKER_BUILDKIT=1 docker build \
+    -f "$DOCKERFILE" . \
+    -t "$TAG" \
+    --iidfile "$ID_FILE"
+else
+  # run build with local docker registery for CI
+  docker pull localhost:5000/"$TAG" || true
```

Comment:
Storing in a local repository with a custom data dir (which is also started by Github Actions) seems like the best balance between build speed and cache speed.